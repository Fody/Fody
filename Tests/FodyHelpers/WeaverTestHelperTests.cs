using System.Collections.Generic;
using System.IO;
using Fody;
using Xunit;

#pragma warning disable 618
public class WeaverTestHelperTests : TestBase
{
    [Fact]
    public void Run()
    {
        var assemblyPath = Path.Combine(CodeBaseLocation.CurrentDirectory, "DummyAssembly.dll");
        var result = new TargetWeaver().ExecuteTestRun(assemblyPath);
    }
}

public class TargetWeaver : BaseModuleWeaver
{
    public override void Execute()
    {
        var type = FindType("System.Boolean");
        Assert.NotNull(type);
    }

    public override IEnumerable<string> GetAssembliesForScanning()
    {
        yield return "mscorlib";
    }
}