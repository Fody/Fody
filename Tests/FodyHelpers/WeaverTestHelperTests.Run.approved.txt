{
  "Errors": [],
  "Messages": [
    {
      "Text": "\tRemoving reference to 'Tests'.",
      "MessageImportance": "Low"
    }
  ],
  "Warnings": [],
  "AssemblyPath": "\fodytemp\DummyAssembly.dll",
  "FullName": "DummyAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null"
}